packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -3824,6 +3824,211 @@ describe("typespec-client-generator-core: types", () => {
+    });
+
+    it("basic arg type", async function () {
+      runner = await createSdkTestRunner({
+        emitterName: "@azure-tools/typespec-python",
+        "decorators-white-list": ["Azure.ClientGenerator.Core.@clientName"],
+      });
+
+      await runner.compileWithBuiltInService(`
+        model Blob {
+          @clientName("ID")
+          id: string;
+        }
+
+        op test(): Blob;
+      `);
+
+      const models = runner.context.experimental_sdkPackage.models;
+      strictEqual(models.length, 1);
+      deepStrictEqual(
+        models[0].properties[0].decorators["Azure.ClientGenerator.Core.@clientName"],
+        { rename: "ID", scope: undefined }
+      );
+    });
+
+    it("enum member arg type", async function () {
+      runner = await createSdkTestRunner({
+        emitterName: "@azure-tools/typespec-python",
+        "decorators-white-list": ["TypeSpec.@encode"],
+      });
+
+      await runner.compileWithBuiltInService(`
+        model Blob {
+          @encode(BytesKnownEncoding.base64url)
+          value: bytes;
+        }
+
+        op test(): Blob;
+      `);
+
+      const models = runner.context.experimental_sdkPackage.models;
+      strictEqual(models.length, 1);
+      deepStrictEqual(models[0].properties[0].decorators["TypeSpec.@encode"], {
+        encoding: "base64url",
+        encodedAs: undefined,
+      });
+    });
+  });
+
+  describe("test of xml for general decorators list", () => {
+    it("@attribute", async function () {
+      runner = await createSdkTestRunner({
+        emitterName: "@azure-tools/typespec-python",
+        "decorators-white-list": ["TypeSpec\\.Xml\\..*"],
+      });
+
+      await runner.compileWithBuiltInService(`
+        model Blob {
+          @attribute id: string;
+        }
+
+        op test(): Blob;
+      `);
+
+      const models = runner.context.experimental_sdkPackage.models;
+      strictEqual(models.length, 1);
+      deepStrictEqual(models[0].properties[0].decorators["TypeSpec.Xml.@attribute"], {});
+    });
+
+    it("@name", async function () {
+      runner = await createSdkTestRunner({
+        emitterName: "@azure-tools/typespec-python",
+        "decorators-white-list": ["TypeSpec\\.Xml\\..*"],
+      });
+
+      await runner.compileWithBuiltInService(`
+        @name("XmlBook")
+        model Book {
+          @name("XmlId") id: string;
+          content: string;
+        }
+
+        op test(): Book;
+      `);
+
+      const models = runner.context.experimental_sdkPackage.models;
+      strictEqual(models.length, 1);
+      deepStrictEqual(models[0].decorators["TypeSpec.Xml.@name"], { name: "XmlBook" });
+      deepStrictEqual(models[0].properties[0].decorators["TypeSpec.Xml.@name"], { name: "XmlId" });
+    });
+
+    it("@ns", async function () {
+      runner = await createSdkTestRunner({
+        emitterName: "@azure-tools/typespec-python",
+        "decorators-white-list": ["TypeSpec\\.Xml\\..*"],
```

Comment:
`"TypeSpec\\.Xml\\..*"` -> Are these files contain xml related decorators? I don't see it in this pr.

Reply:
all xml decorators are under `TypeSpec.Xml`

Reply:
ok, so this is the way to include all the decorators under `TypeSpec.Xml`? Why there are two escape character `\` (previously I thought it is folder)?

Reply:
> ok, so this is the way to include all the decorators under `TypeSpec.Xml`? Why there are two escape character `\` (previously I thought it is folder)?

ts escape for `\`